test/Microsoft.VisualStudio.Editor.Razor.Test/RazorDirectiveCompletionSourceTest.cs

```diff
@@ -0,0 +1,368 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.AspNetCore.Razor.Language.Extensions;
+using Microsoft.AspNetCore.Razor.Language.Legacy;
+using Microsoft.VisualStudio.Language.Intellisense.AsyncCompletion;
+using Microsoft.VisualStudio.Text;
+using Moq;
+using Xunit;
+using Span = Microsoft.VisualStudio.Text.Span;
+
+namespace Microsoft.VisualStudio.Editor.Razor
+{
+    public class RazorDirectiveCompletionSourceTest : ForegroundDispatcherTestBase
```

Comment:
I'm kinda skipping these because I assume it's pretty similar to what we had before yes?

Reply:
Mostly all new actually. The new APIs are now more testable than the old ones Roslyn provided so I expanded our coverage.